packages/netsuite-adapter/src/query.ts

```diff
@@ -87,7 +92,8 @@ export const andQuery = (firstQuery: NetsuiteQuery, secondQuery: NetsuiteQuery):
 })
 
 export const notQuery = (query: NetsuiteQuery): NetsuiteQuery => ({
-  isTypeMatch: () => true,
+  isTypeMatch: typeName => !query.areAllObjectsMatch(typeName),
+  areAllObjectsMatch: typeName => !query.isTypeMatch(typeName),
```

Comment:
I assume that it indicates this area is not covered well with UTs

```suggestion
  isTypeMatch: typeName => !query.isTypeMatch(typeName),
  areAllObjectsMatch: typeName => !query.areAllObjectsMatch(typeName),
```

Reply:
It's on purpose, `isTypeMatch` of `notQuery(query)` returns false only if `query` matches all the instances of `typeName`, i.e., `query.areAllObjectsMatch` is true.